internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala

```diff
@@ -72,13 +72,14 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile with
 
   private class DependencyProcessor(unit: CompilationUnit) {
     private def firstClassOrModuleClass(tree: Tree): Option[Symbol] = {
-      tree foreach {
-        case classOrModule @ ((_: ClassDef) | (_: ModuleDef)) =>
-          val sym = classOrModule.symbol
-          return Some(if (sym.isModule) sym.moduleClass else sym)
-        case _ => ()
+      val maybeClassOrModule = tree find {
+        case ((_: ClassDef) | (_: ModuleDef)) => true
+        case _                                => false
+      }
+      maybeClassOrModule.map { classOrModule =>
+        val sym = classOrModule.symbol
+        if (sym.isModule) sym.moduleClass else sym
```

Comment:
collect over find/map...

Reply:
I think you meant `collectFirst`, however `collectFirst` does not exist for `Tree`

Reply:
Ah, fair enough (and yeah, collectFirst 😳)